### ISSUE TITLE
Can't resize on Linux when window border is set to fixed

### ISSUE THREAD

**Issue by cra0zy** [opentk/opentk#259]
Title says it all, you can't change window size from the code if the border is set to fixed. I did some hackery to get MonoGame window resizing working on Linux: https://github.com/mono/MonoGame/pull/3315/files, just set border to resizable before resizing and afterward restore the border.

**Comment by Frassle** [comment]
I think I've got a fix for this. Interesting oddity that we even expose ClientRectangle though, looks like we always set x and y to 0? I guess that's following WinForms style which does the same.

We also return ClientSize for Width/Height but set ClientSize when setting Width but Bounds when setting Height!? I'll fix that in the pull req as well.

**Comment by cra0zy** [comment]
Houston we have a problem

While fixing another bug, I found that this bug can still occur.

**Comment by Frassle** [comment]
Do you have repro steps?

**Comment by cra0zy** [comment]
It happened during the creation of the window, and it happened randomly.

Happens around 1 in 15 times.

Here is the code:
```c#
using System.Drawing;

using OpenTK;
using OpenTK.Graphics.OpenGL;
using OpenTK.Input;

namespace OpenTKWindow
{
    class MyApplication
    {
        public static void Main()
        {
            var game = new GameWindow();

            game.Width = 800;
            game.Height = 480;
            game.WindowState = WindowState.Normal;
            game.WindowBorder = WindowBorder.Fixed;

            game.Load += (sender, e) =>
                {
                    // setup settings, load textures, sounds
                    game.VSync = VSyncMode.On;
                };

            game.Resize += (sender, e) =>
                {
                    GL.Viewport(0, 0, game.Width, game.Height);
                };

            game.UpdateFrame += (sender, e) =>
                {
                    // add game logic, input handling
                    if (game.Keyboard[Key.Escape])
                    {
                        game.Exit();
                    }

                    if(game.Keyboard[Key.F])
                    {
                        game.Width += 1;
                    }
                };

            game.RenderFrame += (sender, e) =>
                {
                    // render graphics
                    GL.Clear(ClearBufferMask.ColorBufferBit | ClearBufferMask.DepthBufferBit);

                    GL.MatrixMode(MatrixMode.Projection);
                    GL.LoadIdentity();
                    GL.Ortho(-1.0, 1.0, -1.0, 1.0, 0.0, 4.0);

                    GL.Begin(PrimitiveType.Triangles);

                    GL.Color3(Color.MidnightBlue);
                    GL.Vertex2(-1.0f, 1.0f);
                    GL.Color3(Color.SpringGreen);
                    GL.Vertex2(0.0f, -1.0f);
                    GL.Color3(Color.Ivory);
                    GL.Vertex2(1.0f, 1.0f);

                    GL.End();

                    game.SwapBuffers();
                };

            // Run the game at 60 updates per second
            game.Run(60.0);
        }
    }
}
```

**Comment by Frassle** [comment]
Ok so if I compile that with latest develop add a WriteLine(game.Width) after incrementing it in UpdateFrame and hold "f" it prints out something like the following:
```
878
878
878
878
878
878
878
878
878
878
878
878
879
879
879
879
879
879
879
879
879
879
879
879
879
880
880
880
880
880
880
880
880
880
880
880
```

So most of the increments don't actually cause a change. Will look into it.

**Comment by Frassle** [comment]
So seem the issue is that the resize event doesn't get picked up by the ProcessEvents call at the end of setting ClientSize. Can rejig ProcessEvents to block waiting for the expected event to fix this.

**Comment by Nihlus** [comment]
Fixed via #567.
